ibis/backends/bigquery/registry.py

```diff
@@ -161,10 +162,14 @@ def _regex_search(translator, op):
 def _regex_extract(translator, op):
     arg = translator.translate(op.arg)
     regex = _translate_pattern(translator, op.pattern)
-    index = translator.translate(op.index)
-    matches = f"REGEXP_CONTAINS({arg}, {regex})"
-    extract = f"REGEXP_EXTRACT_ALL({arg}, {regex})[SAFE_ORDINAL({index})]"
-    return f"IF({matches}, IF(COALESCE({index}, 0) = 0, {arg}, {extract}), NULL)"
+
+    if op.index is not None:
+        warnings.warn(
```

Comment:
Can you raise an exception here instead?

Reply:
Would ValueError be appropriated?

Reply:
I think `com.OperationNotDefinedError` is appropriate here.

Reply:
The index parameter is marked as required in the [docs](https://ibis-project.org/api/expressions/strings/?h=re_e#ibis.expr.types.strings.StringValue.re_extract). Should I raise an exception and update the docs?

Reply:
It looks like bigquery allows one capture group, so it seems like `0` or `1` would be the allow values for `op.index`. It's probably best to let the database handle this one since this behavior is kind of an outlier.

Reply:
So I think the best option is to warn the user that the index parameter has no effect for BigQuery

Reply:
I see. Ok, I think then the warning is probably the best option.

Thanks for working through this!

Reply:
@tswast Can you take a look at this? Thanks!

Reply:
See my latest comment on the main PR for an alternative that does support `index != 0`.

Reply:
Ugh. I need to correct it for when there aren't matches. The `REGEXP_CONTAINS` logic thankfully doesn't care if there are > 1 matching groups.

```
SELECT 
IF(REGEXP_CONTAINS(my_string, pattern),
  IF(
  re_index = 0,
  REGEXP_EXTRACT(my_string, pattern),  
  REGEXP_REPLACE(my_string, CONCAT(pattern, ".*"), CONCAT("\\", CAST(re_index AS STRING)))
  ),
  NULL
)
FROM UNNEST([
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("123.456.789.nomatch", "([0-9]+)\\.([0-9]+)\\.([0-9]+)", 1),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("123.456.789.nomatch", "([0-9]+)\\.([0-9]+)\\.([0-9]+)", 2),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("123.456.789.nomatch", "([0-9]+)\\.([0-9]+)\\.([0-9]+)", 3),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("nomatch.nomatch.nomatch", "([0-9]+)\\.([0-9]+)\\.([0-9]+)", 3),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("123.456.789.nomatch", "[0-9]+\\.[0-9]+\\.[0-9]+", 0),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("nomatch.123.456.789.nomatch", "[0-9]+\\.[0-9]+\\.[0-9]+", 0),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("a|b|c", "([a-zA-Z0-9]+(?:\\|[a-zA-Z0-9]+){0,1})$", 0),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("b|a|c", "([a-zA-Z0-9]+(?:\\|[a-zA-Z0-9]+){0,1})$", 0),
  STRUCT<my_string STRING, pattern STRING, re_index INT64>("b|b|b|c|a", "([a-zA-Z0-9]+(?:\\|[a-zA-Z0-9]+){0,1})$", 0)
])
```